openjdk.test.modularity/src/com.stf/net/adoptopenjdk/test/modularity/JlinkTest.java

```diff
@@ -0,0 +1,373 @@
+ */
+
+public class JlinkTest  implements StfPluginInterface {
+	
+	private ModuleRef mainModuleMjar;
+	private ModuleRef helloModuleMjar;
+	private ModuleRef mainModuleJmod;
+	private ModuleRef helloModuleJmod;
+	private DirectoryRef nativeDir; 
+	
+	/**
+	 * Each of the values of the following enum corresponds to a particular test.
+	 * In order to locally run a test from JlinkTest suite, provide its corresponding 
+	 * enum value in the command line like: 
+	 * 		stf.pl -test=JlinkTest -test-args="variant=<enum_value_of_test>" 
+	 * */
+	public enum Variant { 
+		RequiredMod,
+		AddModLimitMod
+	};
+	
+	public void help(HelpTextGenerator help) {
+		help.outputSection("Custom runtime test");
+		help.outputText("This test aims at creating custom runtime "
+				+ "images and run the test using the image.");
+	}
+
+	public void pluginInit(StfCoreExtension test) throws StfException {
+	}
+	
+	public void setUp(StfCoreExtension test) throws StfException {
+		
+		DirectoryRef jLinkDir = test.env().findTestDirectory("openjdk.test.modularity/bin/tests/com.test.jlink");
+		DirectoryRef confDir = test.env().findTestDirectory("openjdk.test.modularity/bin/tests/com.test.jlink/conf");
+		nativeDir = test.env().findTestDirectory("openjdk.test.modularity/bin/tests/com.test.jlink/native/bin" + 
+																				"/" + test.env().primaryJvm().getJavaVersionCode() + "/" + test.env().getPlatform());
+		DirectoryRef helloDir = test.env().findTestDirectory("openjdk.test.modularity/bin/common-mods/com.hello");
+		
+		/*Creating a modularized jar / jmod out of com.test.jlink.*/ 
+		 
+		 /*com.test.jlink has native files which are not required, so 
+		 * creating a copy where the unwanted folders are removed.*/
+		DirectoryRef jLinkDirCopy = test.doCpDir("Copying com.test.jlink", jLinkDir, 
+				test.env().getTmpDir().childDirectory("../com.test.jLinkDirCopy"));
+		
+		/*Removing the native folder from com.test.jlink*/
+		test.doRm("Removing natives folder from com.test.jLinkDirCopy", 
+				jLinkDirCopy.childDirectory("native"));
+		
+		/* The resulting jars will be written to the result dir 
+		 * set in the beginning of the run.*/
+		mainModuleMjar = test.doCreateModularJar("Create com.test.jlink jar", "openjdk.test.modularity/bin/tests/com.test.jlink");
+		helloModuleMjar = test.doCreateModularJar("Create com.hello jar", "openjdk.test.modularity/bin/common-mods/com.hello");
+		
+		test.doValidateFileExists("Check com.test.jlink modular jar created", mainModuleMjar.getJarFileRef());
+		test.doValidateFileExists("Check com.hello modular jar created", helloModuleMjar.getJarFileRef());
+		
+		/*Removing the conf folder from com.test.jlink*/
+		test.doRm("Removing conf folder from com.test.jLinkDirCopy", jLinkDirCopy.childDirectory("conf"));
+
+		/* We need to pass <arch>/*.so as input to --lib in jmod. 
+		 * This is required only for non-windows platforms.*/
+		
+		/*Creating the <arch>/*.so structure*/ 
+		if ( ! test.env().getPlatform().contains("win_x86") ) {
```

Comment:
Is this test not executed/valid on Windows?

Reply:
The test should run on Windows too. I remember Mamtha stating, while she was writing this test that, on non-windows platforms, the generated native library file has to be copied over to that specific location. On Windows it is not required. Hence the copy step was done only for no-Windows platforms.